variant-remapping-automation/genome_target_tracker.py

```diff
@@ -0,0 +1,116 @@
+
+remapping_genome_target_table = 'evapro.supported_assembly_tracker'
+ensembl_url = 'http://rest.ensembl.org/info/assembly'
+
+
+def get_all_taxonomies_from_eva(private_config_xml_file):
+    taxonomy_list = []
+    with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+        query = 'SELECT DISTINCT taxonomy_id FROM evapro.taxonomy'
+        for taxonomy in get_all_results_for_query(pg_conn, query):
+            taxonomy_list.append(taxonomy[0])
+
+    return taxonomy_list
+
+
+def get_tax_asm_from_eva(private_config_xml_file):
+    eva_tax_asm = {}
+    with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+        query = f"""SELECT DISTINCT taxonomy_id, source, assembly_id FROM {remapping_genome_target_table} 
+        WHERE current=TRUE"""
+        for tax_id, source, assembly in get_all_results_for_query(pg_conn, query):
+            eva_tax_asm[tax_id] = {'assembly': assembly, 'source': source}
+
+    return eva_tax_asm
+
+
+def get_tax_asm_from_sources(taxonomy_list):
+    source_tax_asm = {}
+    for tax_id in taxonomy_list:
+        # Get tax_asm from Ensembl
+        tax_asm_from_ensembl = get_tax_asm_from_ensembl(tax_id)
+        if tax_asm_from_ensembl is not None:
+            source_tax_asm[tax_id] = tax_asm_from_ensembl
+
+    return source_tax_asm
+
+
+def get_tax_asm_from_ensembl(tax_id):
+    try:
+        logger.info(f'Query Ensembl for species name using taxonomy {tax_id}')
+        sp_name = get_scientific_name_from_ensembl(tax_id)
+    except Exception:
+        logger.warning(f'Could not get species name for taxonomy {tax_id} in Ensembl')
+        return None
+    # Get assembly from Ensembl
+    logger.info(f'Query Ensembl for supported assembly for taxonomy {tax_id} using species_name "{sp_name}"')
+    assembly = get_supported_asm_from_ensembl(sp_name)
+    if assembly != 'None':
+        return {'assembly': assembly, 'source': 'Ensembl'}
+    else:
+        logger.warning(
+            f'Could not find supported assembly for taxonomy_id {tax_id} using species_name "{sp_name}" in Ensembl')
+        return None
+
+
+def get_supported_asm_from_ensembl(scientific_name):
+    url = ensembl_url + '/' + scientific_name.lower().replace(' ', '_')
+    response = requests.get(url, params={'content-type': 'application/json'})
+    data = response.json()
+    assembly_accession = str(data.get('assembly_accession'))
+    return assembly_accession
+
+
+def check_supported_target_assembly(private_config_xml_file):
+    taxonomy_list = get_all_taxonomies_from_eva(private_config_xml_file)
+    eva_tax_asm = get_tax_asm_from_eva(private_config_xml_file)
+    source_tax_asm = get_tax_asm_from_sources(taxonomy_list)
+
+    taxonomy_not_tracked_by_eva = []
+    taxonomy_not_present_in_sources = []
+
+    for tax_id in taxonomy_list:
+        if tax_id in eva_tax_asm and tax_id in source_tax_asm:
+            if eva_tax_asm[tax_id] != source_tax_asm[tax_id]:
```

Comment:
```suggestion
            if eva_tax_asm[tax_id]['assembly'] != source_tax_asm[tax_id]['assembly']:
```

Reply:
Actually, all of this needs to be made source-specific since we've added that as a field - i.e. we should warn about a mismatch for Ensembl only for that source, etc.

It might make the bookkeeping a bit easier to make a simple assembly object with taxonomy id, assembly accession, and source as properties rather than passing around a dict, but that's up to you of course.

Reply:
done